api/src/main/java/uk/gov/dhsc/htbhf/claimant/reporting/ClaimantCategoryCalculator.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.dhsc.htbhf.claimant.reporting;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import uk.gov.dhsc.htbhf.claimant.entitlement.PregnancyEntitlementCalculator;
+import uk.gov.dhsc.htbhf.claimant.entity.Claimant;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.Period;
+import java.util.List;
+
+import static uk.gov.dhsc.htbhf.claimant.reporting.payload.ClaimantCategory.NOT_PREGNANT_WITH_CHILDREN;
+import static uk.gov.dhsc.htbhf.claimant.reporting.payload.ClaimantCategory.PREGNANT_AND_UNDER_16;
+import static uk.gov.dhsc.htbhf.claimant.reporting.payload.ClaimantCategory.PREGNANT_WITH_CHILDREN;
+import static uk.gov.dhsc.htbhf.claimant.reporting.payload.ClaimantCategory.PREGNANT_WITH_NO_CHILDREN;
+
+@Component
+@RequiredArgsConstructor
+public class ClaimantCategoryCalculator {
```

Comment:
Looking at this, I was wondering if it could mostly be encapsulated within an Enum which would simplify this greatly. From looking at it we would need initially calculate three values (ageInYears, isClaimantPregnant and hasChildren) and simply lookup the enum value based from those answers (which would be in the enum). What do you think?